pkg/controller/topology.go

```diff
@@ -279,16 +312,13 @@ func GenerateAccessibilityRequirements(
 // getSelectedCSINode returns the CSINode object for the given selectedNode.
 func getSelectedCSINode(
 	csiNodeLister storagelistersv1.CSINodeLister,
-	selectedNode *v1.Node) (*storagev1.CSINode, error) {
+	selectedNodeName string) (*storagev1.CSINode, error) {
 
-	selectedCSINode, err := csiNodeLister.Get(selectedNode.Name)
+	selectedCSINode, err := csiNodeLister.Get(selectedNodeName)
 	if err != nil {
 		// We don't want to fallback and provision in the wrong topology if there's some temporary
 		// error with the API server.
-		return nil, fmt.Errorf("error getting CSINode for selected node %q: %v", selectedNode.Name, err)
-	}
-	if selectedCSINode == nil {
```

Comment:
Should we keep this check to retain original behavior?

Reply:
The Get call does not return nil, it returns an error is the csinode is not there?

Reply:
Hmm actually there may still be a chance that the csinode as the value in the map can be nil, I was mistaken it for en non-exist key. Added the logic back.